src/engine/V3Eval.v3

```diff
@@ -823,6 +861,19 @@ def do_v_v_x16(a: (u64, u64), f: (u8) -> u8) -> (u64, u64) { // Performs a 16-la
 	}
 	return (low, high);
 }
+// f(v128, int) -> v128
+def do_vi_v_x16(a: (u64, u64), b: i32, f: (u8, u8) -> u8) -> (u64, u64) {
```

Comment:
These are going to do a partial application for each invocation. Can you find a way to refactor so that the partial applications only happen when initializing the publicly-exposed methods?

Reply:
Yes I realized there will be extra runtime overheads if I do things this way. But I haven't figured out an elegant way to refactor the code. For now I change the helpers such as `do_vi_v_x16` to handle the logic themselves to avoid partial applications. Will keep on thinking about how to improve while I'm working on other instructions.